tools/jdbc/src/main/java/org/duckdb/DuckDBResultSet.java

```diff
@@ -115,6 +112,37 @@ private void check(int columnIndex) throws SQLException {
 
 	}
 
+	/**
+	 * Export the result set as an ArrowReader
+	 *
+	 * @param arrow_buffer_allocator an instance of {@link org.apache.arrow.memory.BufferAllocator}
+	 * @param arrow_batch_size number of DuckDB vectors to return (should be multiple of 2)
+	 * @return an instance of {@link org.apache.arrow.vector.ipc.ArrowReader}
+	 */
+	public Object arrowExportStream(Object arrow_buffer_allocator, long arrow_batch_size) throws SQLException {
+		if (isClosed()) {
+			throw new SQLException("Result set is closed");
+		}
+
+		try {
+			Class<?> buffer_allocator_class = Class.forName("org.apache.arrow.memory.BufferAllocator");
+			if (!buffer_allocator_class.isInstance(arrow_buffer_allocator)) {
+				throw new RuntimeException("Need to pass an Arrow BufferAllocator");
+			}
+			Long stream_pointer = DuckDBNative.duckdb_jdbc_arrow_stream(result_ref, arrow_batch_size);
+			Class<?> arrow_array_stream_class = Class.forName("org.apache.arrow.c.ArrowArrayStream");
+			Object arrow_array_stream = arrow_array_stream_class.getMethod("wrap", long.class).invoke(null, stream_pointer);
+
+			Class<?> c_data_class = Class.forName("org.apache.arrow.c.Data");
+
+			return c_data_class
+					.getMethod("importArrayStream", buffer_allocator_class, arrow_array_stream_class)
+					.invoke(null, arrow_buffer_allocator, arrow_array_stream);
+		} catch (Exception e) {
+			throw new RuntimeException(e);
```

Comment:
This is now going to wrap all thrown exceptions in a `RuntimeException`, which is an pain to handle in consuming code. You can probably just move the `duckdb_jdbc_arrow_stream` call outside the try-catch, and that will fix this, at the cost of not type-checking the allocator beforehand

Reply:
Alternatively, you can just be more specific and only catch the reflection exceptions here